lib/crawler.js

```
@@ -25,44 +25,33 @@ var QUEUE_ITEM_INITIAL_DEPTH = 1;
 /*
     Public: Constructor for the crawler.
 
-    host                -   Initial hostname/domain to begin crawling from. By
-                            default, the crawl will be locked to this hostname.
-    initialPath         -   Initial path to begin crawling from.
-    initialPort         -   Port to begin crawling from.
-    interval            -   Request interval for the crawler. Defaults to 250ms.
+    initialURL - Initial URL to crawl. Protocol, host and port settings that
+                 affect the entire crawl are inferred from this URL.
 
     Examples
 
-        var crawler = new Crawler("example.com","/",80,500);
-
-        var crawler = new Crawler("example.com");
+        var crawler = new Crawler("https://example.com/test.html");
 
     Returns the crawler object which has now been constructed.
 
 */
-var Crawler = function(host, initialPath, initialPort, interval) {
-    var crawler = this;
-
-    // Data integrity checks
-    if (initialPort && isNaN(initialPort)) {
-        throw new Error("Port must be a number!");
-    }
+var Crawler = function(initialURL) {
```

Let's throw an error here if we get arguments.length > 1.